macros/sql_gen/gen_final_cte.sql

```diff
@@ -38,12 +58,33 @@
 
         select * from final 
 
+        -- metric where clauses...
+        {% if where %}
+        where (
+            {%- for filter in where %}
+                {{ filter }}
+                {% if not loop.last %} and {% endif %}
+            {%- endfor %}
+        )
+        {% endif %}
+
         {% else %}
 
         -- single metric without secondary calculations
 
         select * from {{base_set[0]}}__final 
 
+
+        -- metric where clauses...
+        {% if where %}
+        where (
+            {%- for filter in where %}
+                {{ filter }}
+                {% if not loop.last %} and {% endif %}
+            {%- endfor %}
+        )
+        {% endif %}
+
```

Comment:
This is another place where your code has to branch based on whether it's a single instance or not, and so you have duplicated code. 

The band-aid solution would be to have 
```
final as ( 
{% if secondary_calcs %} 
select * from secondary_calculations
{% else %}
select * from {{base_set[0]}}__final
{% endif %}
)
```
which would then mean you could `select * from final where [whatever]` without copy-pasting that logic. 

A more involved approach would likely involve having more stuff not care whether it's got one element or many (as we discussed a couple of week back), but that’s out of scope for this PR